codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -260,7 +260,7 @@ public ChannelFuture writeSettings(ChannelHandlerContext ctx, Http2Settings sett
             ChannelPromise promise) {
         outstandingLocalSettingsQueue.add(settings);
         try {
-            if (connection.isGoAway()) {
+            if (connection.goAwayReceived() || connection.goAwaySent()) {
```

Comment:
@nmittler Yes. I have removed the above checks in my GOAWAY PR anyways, as they shouldn't be there. Generally, I can't think of a place where checking for `goAwaySent` and `goAwayReceived` makes sense. Usually, we really have to distinguish between the two.